File: Implementations/Reference/Runtime/rust/src/execute.rs
use std::path::{Path, PathBuf};

use serde_json::Value;

use crate::contract::{default_contract_path, default_wfrog_path};
use crate::diagnostics::Result;
use crate::runtime::RuntimeCore;

pub fn execute_contract(
    input_value: u16,
    contract_path: Option<PathBuf>,
    wfrog_path: Option<PathBuf>,
) -> Result<Value> {
    let contract = contract_path.unwrap_or(default_contract_path()?);
    let wfrog = wfrog_path.unwrap_or(default_wfrog_path()?);
    let mut runtime = RuntimeCore::from_paths(contract, wfrog)?;
    runtime.execute(Some(input_value))
}

File: Implementations/Reference/Runtime/cpp/src/execute.cpp
#include "execute.hpp"

#include "runtime.hpp"

namespace frog::runtime {

frog::json::Value execute_contract(
    std::uint16_t input_value,
    std::optional<std::filesystem::path> contract_path,
    std::optional<std::filesystem::path> wfrog_path) {
    Slice05RuntimeCore runtime(
        contract_path.value_or(default_contract_path()),
        wfrog_path.value_or(default_wfrog_path()));
    return runtime.execute(input_value);
}

} // namespace frog::runtime
